### PR TITLE
Add log_key_prefix to Trainer to control the prefix for metrics like epoch

### PR DESCRIPTION
**Pull request by pupeno**
## What does this PR do?

Contributes to #21782

I made a quick video overview that might save you some time during review: https://youtu.be/TFY9_kAvv5Q

Adds `log_key_prefix` to `Trainer` to let users control the metric keys that the Trainer
generates automatically, without touching user-provided metric names.

Currently the only Trainer-generated key is `epoch`, which is injected into every logging step
via the logger connector. With this change, `Trainer(log_key_prefix="trainer/")` causes that key
to be logged as `trainer/epoch` instead.

The prefix applies **only** to keys generated by the Trainer itself. Keys logged by the user via
`self.log(...)` are never affected.

Default is `None`/`""`, which preserves the previous bare `epoch` behaviour — fully
backward-compatible, no existing behaviour changes.

The injection point is a single `setdefault` call in `_LoggerConnector.update_eval_epoch_metrics`,
so every logger downstream benefits automatically without any logger-side changes.

```python
# before: always logs as "epoch"
trainer = Trainer(...)

# after: opt-in prefix
trainer = Trainer(log_key_prefix="trainer/")  # logs as "trainer/epoch"
```

Sample code can be found on https://github.com/pupeno/pytorch-lightning-workspace/tree/main/21784_trainer_log_key_prefix

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by deependujha — APPROVED** [review]

**Comment by pupeno** [comment]
I updated this to master and there's sample code that uses here: https://github.com/pupeno/pytorch-lightning-workspace/blob/main/scratch/21784/experiment.py

Anything else I can do here to get it merged?

**Comment by pupeno** [comment]
Hey @carmocca, @awaelchli, @Borda, would you be the right people to review this one? Or do you know who would? Thank you.

**Comment by Borda** [comment]
> Hey @carmocca, @awaelchli, @Borda, would you be the right people to review this one? Or do you know who would? Thank you.

Thank you for reaching out, unfortunately you need to ping maintainers or code owners...

**Comment by pupeno** [comment]
Would @lantiga, @tchaton, @justusschock, @ethanwharris be the right people to review and/or approve then?

**Comment by pupeno** [comment]
@bhimrazy do you know who would be the right person to review this PR? I created a short video that just walks the reviewer through the code, so hopefully it should be super quick: https://youtu.be/TFY9_kAvv5Q

**Review by ethanwharris — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by bhimrazy — APPROVED** [review]
